lib/devices.ts

```diff
@@ -105,6 +110,121 @@ export class RaspberryPi extends DeviceInteractor {
+	}
+
+	async flash(stream: Stream.Readable) {
+		// await this.testBot.flash(stream);
+		console.log('Flash method for Fin');
+
+		await this.powerOnFlash();
+		// etcher-sdk (power on) usboot
+		const adapters: sdk.scanner.adapters.Adapter[] = [
+			new sdk.scanner.adapters.BlockDeviceAdapter(() => false),
+			new sdk.scanner.adapters.UsbbootDeviceAdapter(),
+		];
+		const deviceScanner = new sdk.scanner.Scanner(adapters);
+		console.log('Waiting for compute module');
+		// Wait for compute module to appear over usb
+		const computeModule: sdk.sourceDestination.UsbbootDrive = await new Promise(
+			(resolve, reject) => {
+				function onAttach(
+					drive: sdk.scanner.adapters.AdapterSourceDestination,
+				) {
+					if (drive instanceof sdk.sourceDestination.UsbbootDrive) {
+						deviceScanner.removeListener('attach', onAttach);
+						resolve(drive);
+					}
+				}
+				deviceScanner.on('attach', onAttach);
+				deviceScanner.on('error', reject);
+				deviceScanner.start();
+			},
+		);
+		console.log('Compute module attached');
+		// wait to convert to block device.
+		await new Promise<void>((resolve, reject) => {
+			function onDetach(drive: sdk.scanner.adapters.AdapterSourceDestination) {
+				if (drive === computeModule) {
+					deviceScanner.removeListener('detach', onDetach);
+					resolve();
+				}
+			}
+			deviceScanner.on('detach', onDetach);
+			deviceScanner.on('error', reject);
+		});
+
+		console.log('Waiting for compute module to reattach as a block device');
+
+		const dest = await new Promise(
+			(resolve: (drive: sdk.sourceDestination.BlockDevice) => void, reject) => {
+				function onAttach(
+					drive: sdk.scanner.adapters.AdapterSourceDestination,
+				) {
+					if (
+						drive instanceof sdk.sourceDestination.BlockDevice &&
+						drive.description === 'Compute Module'
+					) {
+						console.log('Attached compute module.');
+						resolve(drive);
+						deviceScanner.removeListener('attach', onAttach);
+					}
+				}
+				deviceScanner.on('attach', onAttach);
+				deviceScanner.on('error', reject);
+			},
+		);
+		deviceScanner.stop();
+
+		await Bluebird.delay(1000); // Wait 1s before trying to flash
+		// flash
+		console.log('About to attempt flashing');
+		await this.testBot.flashToDisk(dest, stream);
+
+		console.log('Flashed!');
+		await this.toggleUsb(false, 4);
+		await this.testBot.powerOffDUT();
+	}
+
+	async powerOn() {
+		console.log('Powering on Fin');
+		await this.toggleUsb(false, 4);
+		// await Bluebird.delay(5 * 1000);
+		await this.testBot.setVout(this.powerVoltage);
+		await this.testBot.powerOnDUT();
+	}
+}
+
+/** Implementation for balenaFin v1.0.0 (V09) */
+export class BalenaFinV09 extends BalenaFin {
+	async powerOnFlash() {
+		await this.toggleUsb(false, 4);
+		await Bluebird.delay(1000); // Wait 1s before trying to turning USB back on
+		await this.toggleUsb(true, 4);
+		await this.testBot.setVout(this.powerVoltage);
+		await this.testBot.powerOnDUT();
+	}
```

Comment:
@Bucknalla Regarding `BalenaFinV09` class, can you explain bit more on the poweronFlash() method and the need for a special function that bends the format. Usually, each class should be having only 3 functions, ON, OFF, and flash.

Reply:
@vipulgupta2048 changing this to a protected class, as its only used internally to ensure the CM3 powers up in the correct sequence to enter USB boot mode